### DISCUSSION TITLE
Is it possible to use a ML model trained in MATLAB/Python during runtime in OMNet++ ?

### DISCUSSION THREAD

**Discussion by sourav003** [omnetpp/omnetpp#1197] [Q&A]
Hi
I need to use some ML based predictive methods in OMNeT++. 
So, it is possible to integrate MATLAB/Python to import any trained models from there and use it in the runtime?
Any helps or pointers will be greatly appreciated.

Thanks in advance.

**Comment by aarizaq** [comment]
It is possible to link python with omnet using sockets. In this case, the python must work as server. Omnet will open a socket with python and will request something, the simulation will be stopped until the server return the data.

**Comment by torokati44** [comment]
An other option would be integrating something like frugally-deep (https://github.com/Dobiasd/frugally-deep) into your simulation directly. Its purpose is exactly to make deep learning models trained in Python directly usable for inference in C++ applications.
